rocket/src/main/scala/ipxact/IpxactGeneratorTransform.scala

```diff
@@ -0,0 +1,419 @@
+      Some(
+        <spirit:parameters>
+          {parametersXml}
+        </spirit:parameters>
+      )
+    }
+  }
+
+  /**
+    * construct the address section from the annotation
+    * @param annotation  address map information
+    * @return
+    */
+  def addressAnnotationsToXml(annotation: AddressMapAnnotation): Seq[scala.xml.Node] = {
+    annotation.mapping.map { case AddressMapEntry(range, permissions, names) =>
+      ???
+    }
+  }
+
+  /**
+    * builds the register map stuff from the given annotation
+    * @param annotation reg field information
+    * @return
+    */
+  def regFieldDescMappingAnnotationsToXml(name: String, annotation: RegFieldDescMappingAnnotation): Seq[Any] = {
+    <spirit:memoryMap>
+      <spirit:name>{s"${name}_mm"}</spirit:name>
+      <spirit:addressBlock>
+        <spirit:name>{annotation.regMappingSer.displayName}</spirit:name>
+        <spirit:baseAddress>{annotation.regMappingSer.baseAddress}</spirit:baseAddress>
+        <spirit:range>{annotation.regMappingSer.regFields.length}</spirit:range>
+        <spirit:width>{Ipxact.MemoryMapWidth}</spirit:width>
+          {
+            annotation.regMappingSer.regFields.map { regField =>
+              <spirit:register>
+                <spirit:name>{regField.name}</spirit:name>
+                <spirit:addressOffset>{regField.byteOffset}</spirit:addressOffset>
+                <spirit:size>{regField.bitWidth}</spirit:size>
+              </spirit:register>
+          }
+        }
+      </spirit:addressBlock>
+    </spirit:memoryMap>
+  }
+
+  //scalastyle:off cyclomatic.complexity
+  def generateMemoryMaps(annotationSeq: AnnotationSeq, moduleName: String): Option[scala.xml.Node] = {
+    val memoryMapSections = annotationSeq.flatMap {
+      case a: RegFieldDescMappingAnnotation =>
+        regFieldDescMappingAnnotationsToXml(moduleName, a)
+      case _ =>
+        Seq.empty
+    }
+
+
+    if(memoryMapSections.isEmpty) {
+      None
+    }
+    else {
+      Some(
+        <spirit:memoryMaps>
+          {memoryMapSections}
+        </spirit:memoryMaps>
+
+      )
+    }
+  }
+
+
+  //TODO: (chick) Should this module be module specific
+  def generateModel(state: CircuitState, moduleName: String): Seq[scala.xml.Node] = {
+    <spirit:model>
+      <spirit:views>
+        <spirit:view>
+          <spirit:name>RTL</spirit:name>
+          <spirit:envIdentifier>::</spirit:envIdentifier>
+          <spirit:fileSetRef>
+            <spirit:localName>hdlSource</spirit:localName>
+          </spirit:fileSetRef>
+        </spirit:view>
+      </spirit:views>
+        {
+          getPorts(state, moduleName)
+        }
+    </spirit:model>
+  }
+
+  /**
+    * Describes where the associated verilog source resides
+    * @param moduleName module name being rendered
+    * @return
+    */
+  //TODO: (chick) how does this related to one-file per module and the name of the verilog output in general
+  // can we just point to all .v files for each module
```

Comment:
I think this bears some thinking about. I think we may want to have a mechanism for transforms to find out where generated rtl ends up. Also useful for generating build system stuff, perhaps @jwright6323 has thoughts about such a mechanism.